test/skimNtuple_HHbtag.cpp

```diff
@@ -1038,172 +1046,211 @@ int main (int argc, char** argv)
+	      vH2.SetPxPyPzE (theBigTree.genpart_px->at(idx2), theBigTree.genpart_py->at(idx2), theBigTree.genpart_pz->at(idx2), theBigTree.genpart_e->at(idx2) );
+	      theSmallTree.m_genH1_pt = vH1.Pt();
+	      theSmallTree.m_genH1_eta = vH1.Eta();
+	      theSmallTree.m_genH1_phi = vH1.Phi();
+	      theSmallTree.m_genH1_e = vH1.E();
+	      theSmallTree.m_genH2_pt = vH2.Pt();
+	      theSmallTree.m_genH2_eta = vH2.Eta();
+	      theSmallTree.m_genH2_phi = vH2.Phi();
+	      theSmallTree.m_genH2_e = vH2.E();
+	      vSum = vH1+vH2;
+	      mHH = vSum.M();
+	      vH1.Boost(-vSum.BoostVector());
+	      ct1 = vH1.CosTheta();
+	      
+	      
+	      // FRA DEBUG - build gen b jets
+	      if (idx1hs_b != -1 && idx2hs_b != -1)
+		{
+		  vGenB1.SetPxPyPzE (theBigTree.genpart_px->at(idx1hs_b), theBigTree.genpart_py->at(idx1hs_b), theBigTree.genpart_pz->at(idx1hs_b), theBigTree.genpart_e->at(idx1hs_b) );
+		  vGenB2.SetPxPyPzE (theBigTree.genpart_px->at(idx2hs_b), theBigTree.genpart_py->at(idx2hs_b), theBigTree.genpart_pz->at(idx2hs_b), theBigTree.genpart_e->at(idx2hs_b) );
 		}
+	      else {
+		cout << "** WARNING: couldn't find 2 H->bb gen dec prod " << idx1hs_b << " " << idx2hs_b << endl;
+	      }
+	      
+	      if (HHrewType == kDiffRew)      HHweight = hhreweighter->getWeight(mHH, ct1);
+	      else if (HHrewType == kC2scan)  HHweight = hhreweighter->getWeight(mHH, ct1, c2_rew);
+	      else if (HHrewType == kOverRew) HHweight = hhreweighter->getWeight(mHH, ct1, kl_rew, kt_rew, c2_rew, cg_rew, c2g_rew);
+	      
+	      theSmallTree.m_genMHH = mHH;
+	      theSmallTree.m_genCosth = ct1;
+	      
+	      // cout << " ........... GEN FINISHED ........... " << " evt=" << theBigTree.EventNumber << " run=" << theBigTree.RunNumber << " lumi=" << theBigTree.lumi << endl;
+	    } // if isHHsignal
+
+
+	  // storing H or Z decaying into bb in dedicated vectors
+	  std::vector<TLorentzVector> higgsVectors;
+	  std::vector<TLorentzVector> bQuarksFromHiggs;
+	  
+	  std::vector<TLorentzVector> zBosonVectors;  
+	  std::vector<TLorentzVector> bQuarksFromZ;
+
+	  
+	  for (unsigned int igen = 0; igen < theBigTree.genpart_px->size(); igen++) {
+	    int pdg = theBigTree.genpart_pdg->at(igen);
+	    bool isQuarkb = (pdg == 5);
+	    TLorentzVector vBoson, vB1;
+	    int idx_H = -1.;
+	    int idx_Z = -1.;
+
+	    if(isQuarkb){ //focusing only on b (not anti-b), assuming that a b from a Z or a Higgs is paired with an anti-b
+	      idx_H = theBigTree.genpart_HMothInd->at(igen);
+	      idx_Z = theBigTree.genpart_ZMothInd->at(igen);
+	      if (idx_H != -1.) { // b from H
+		vBoson.SetPxPyPzE(theBigTree.genpart_px->at(idx_H), theBigTree.genpart_py->at(idx_H), theBigTree.genpart_pz->at(idx_H), theBigTree.genpart_e->at(idx_H));
+		vB1.SetPxPyPzE(theBigTree.genpart_px->at(igen), theBigTree.genpart_py->at(igen), theBigTree.genpart_pz->at(igen), theBigTree.genpart_e->at(igen));
+		higgsVectors.push_back(vBoson);  
+		bQuarksFromHiggs.push_back(vB1);
```

Comment:
related to the question that was not clear, this is where it came from. Here you're saving both the Higgs 4vec and the b (not anti-b) 4vec. For some reason I read that as if you were only saving the b quark, which triggered the question. But in fact that question made no sense so you can ignore it.

However new question: what is this `bQuarksFromHiggs` vector used for?

Reply:
When I implemented gen matching, I later did some post-skim checks. For this reason, I saved some things that might not be necessary in the final version... If we decide we don't need them, I can delete them!